packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -1820,7 +1831,11 @@ function commitRoot(root) {
   return null;
 }
 
-function commitRootImpl(root, renderPriorityLevel) {
+function commitRootImpl(
+  root: FiberRoot,
+  recoverableErrors: null | Array<mixed>,
```

Comment:
This is kind of annoying but there's no obvious place to store these. We don't usually pass things directly from the render phase to the commit phase because we stash it on a fiber.

Reply:
I don't get this part. All calls to commitRoot call this with the global `workInProgressRootRecoverableErrors` which you have access to here too. Other `workInProgress` globals are accessed here too.

So can't you just avoid passing it and instead access `workInProgressRootRecoverableErrors` here?

Reply:
There's an annoying edge case when both 1) you're in a multi-root app 2_ the commit phase is delayed with `setTimeout` (this used to be more common but now it's really just the Suspense train model). During the timeout delay, the next root can start rendering. So you can't access work loop variables directly; you need to stash data somewhere in the fiber tree or on the FiberRoot.

A ren

Reply:
Ah yes, and with suspense images probably. That *is* annoying.